examples/elf/tests/pthread/pthread.c

```diff
@@ -82,16 +82,16 @@ enum exit_values_e
 
 void *child_start_routine(void *arg)
 {
-  printf("CHILD: started with arg=%d\n", (int)arg);
+  printf("CHILD: started with arg=%d\n", (intptr_t)arg);
 
```

Comment:
This will cause warnings in some environments where there is type checking (like sim).  It can also cause parateter mismatches in other cases.

For example, on 64-bit simulation, %d will expect an int argument and sizeof(int) == 32-bits.  But intprt_t will be 64-bits so the printf will be wrong and, in certain cases, can lead to a crash.

My suggesting is to use the more complex cast (int)((intptr_t)arg).  That will both avoid the warning and assure that the value provided to printf is a proper integer type.